tests/crowdsourcing/test_analysis.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import unittest
+from typing import Any, Dict, List
+from parlai.crowdsourcing.utils.analysis import AbstractResultsCompiler
+from collections import defaultdict
+
+try:
+    from mephisto.data_model.unit import Unit
+
+    class MockUnit(Unit):
+        pass
+
+
+except ModuleNotFoundError:
+
+    class MockUnit:
+        def __init__(self, *args, **kwargs) -> None:
+            pass
+
+
+######################################################################
+#
+#           Mock version of Mephisto abstarctions
+#
+######################################################################
+
+
+DUMMY_OPT = {'task_name': 'mock_task_name', 'output_folder': '/dummy/tmp'}
+LEN_MOCK_DATA = 5
+
+
+class MockMephistoDB:
+    def optimized_load(self, *args, **kwargs):
+        return kwargs.get("row")
+
+    def cache_result(self, cls, loaded_val):
+        pass
+
+    def get_worker_name(self, worker_id: str):
+        return {'worker_name': f'UNITTEST_MOCK_{worker_id}'}
+
+
+class MockMephistoBrowser:
```

Comment:
Really great to have this - this'll make testing of future crowdsourcing analysis scripts much easier :)